File: Java/SOS/Node.java
import java.util.ArrayList;
import java.util.List;

/**
 * Node
 */
public class Node {

    public List<Integer> CurrentSubset;
    public int Index;
    public int Sum;

    public Node CreateChildNode() {
        Node child = new Node();
        child.CurrentSubset = new ArrayList<Integer>(CurrentSubset);
        child.Index = Index + 1;
        child.Sum = Sum;
        return child;
    }

    public Node CreateChildNode(int Element) {
        Node child = CreateChildNode();
        child.CurrentSubset.add(Element);
        child.Sum += Element;
        return child;
    }
}

File: SumOfSubset/Node.cs
using System.Collections.Generic;

namespace SumOfSubset
{
    class Node
    {
        public int Index { get; set; }
        public List<int> CurrentSubset { get; set; }

        public Node CreateChildNode(int? Element)
        {
            var child = new Node() { Index = this.Index + 1, CurrentSubset = new List<int>(this.CurrentSubset) };
            if (Element.HasValue)
            {
                child.CurrentSubset.Add(Element.Value);
            }
            return child;
        }
    }
}
